### PR TITLE
Presumably better buffer size for reading from fs

### PR DESCRIPTION
**Pull request by nickitat**
### Changelog category (leave one):
- Performance Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Default size of a read buffer for reading from local filesystem changed to a slightly better value. Also two new settings are introduced: `max_read_buffer_size_local_fs` and `max_read_buffer_size_remote_fs`.

### CODE REVIEW

**Comment by nickitat** [comment]
I can reproduce around 2 times speedup for 

0.070 | 0.023 | -2.941x | -0.660 | 0.503 | order_by_read_in_order | 0 | SELECT * FROM hits_100m_single ORDER BY (CounterID, EventDate) LIMIT 100
-- | -- | -- | -- | -- | -- | -- | --
1.439 | 0.730 | -1.971x | -0.493 | 0.492 | read_hits_with_aio | 1 | SELECT count() FROM hits_100m_single where EventDate between toDate('2013-07-10') and toDate('2013-07-16') and UserID=123 SETTINGS max_threads = 1, min_bytes_to_use_direct_io = 1, max_read_buffer_size = 10485760;

Totals also look ok: -1.006x, -1.009x, -1.008x, +1.002x.

let's rerun tests to also check arm perf tests

btw, justification for 128kb could be found in this [post](https://eklitzke.org/efficient-file-copying-on-linux)

**Comment by robot-ch-test-poll1** [comment]
This is an automated comment for commit 0be9c0751eaf13aec27759436e314abbbb247877 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49321/0be9c0751eaf13aec27759436e314abbbb247877/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Comment by alexey-milovidov** [comment]
Waiting for changelog.

File: src/IO/ReadSettings.h [hunk 1]
```diff
@@ -68,7 +68,7 @@ struct ReadSettings
     /// Method to use reading from remote filesystem.
     RemoteFSReadMethod remote_fs_method = RemoteFSReadMethod::threadpool;
 
-    size_t local_fs_buffer_size = DBMS_DEFAULT_BUFFER_SIZE;
+    size_t local_fs_buffer_size = 128 * 1024;
```
Comment (alexey-milovidov) [inline src/IO/ReadSettings.h:71]: Add a comment here, and include the link as well.

File: src/Interpreters/Context.cpp [hunk 2]
```diff
@@ -4288,7 +4288,7 @@ ReadSettings Context::getReadSettings() const
             "Invalid value '{}' for max_read_buffer_size", settings.max_read_buffer_size);
     }
 
-    res.local_fs_buffer_size = settings.max_read_buffer_size;
+    /* res.local_fs_buffer_size = settings.max_read_buffer_size; */
```
Comment (alexey-milovidov) [inline src/Interpreters/Context.cpp:4291]: Remove it altogether.

Comment (nickitat) [reply src/Interpreters/Context.cpp:4291]: maybe it worth to create a separate setting `max_read_buffer_size_for_local_fs`?

Comment (alexey-milovidov) [reply src/Interpreters/Context.cpp:4291]: Yes, let's introduce two new settings - for local and remote fs, and the old max_read_buffer_size setting will be used if they are unset.

**Comment by nickitat** [comment]
tested on CB to look onto results with real disk, looks ok

Screenshot 2023-05-02 at 00 51 36

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**